### DISCUSSION TITLE
how to specify version when using path dependencies?

### DISCUSSION THREAD

**Discussion by dongho-jung** [python-poetry/poetry#6525] [Q&A] [answered]
With [path dependencies](https://python-poetry.org/docs/dependency-specification/#path-dependencies) 
I can specify dependencies located in a local directory as follows:

```
[tool.poetry.dependencies]
# directory
my-package = { path = "../my-package/", develop = false }

# file
my-package = { path = "../my-package/dist/my-package-0.1.0.tar.gz" }
```

However this can be problem when caching, because modifying local dependencies won't affect lock file.
if only it could be versioned, I can modify my local file and update the version within dependant packages so that it can make use of cache with updated local dependencies.

can I? or if I can't, how do you manage your pyproject.toml with path dependencies for caching?

**Comment by neersighted** [comment, marked as answer]
The kind of caching you are talking about is unclear, but by the very nature of local dependencies they are unversioned as you are not asking Poetry to select and fetch a version for you -- you are supplying a package directly to Poetry to be installed into your project environment.

If your local dep is in the container build context then the `COPY` instruction that brings it in should be invalidated by the files within changing. For other forms of caching, specifics would be needed to give any sort of advice.

**Reply by dongho-jung** [reply]
On second thoughts, your answer is absolutely right. I'm supplying a package directly Poetry indeed. Unfortunately, I'm not using docker now so I can't make use of that. I think I gotta find a way to specify the changes of local files. thanks
